### DISCUSSION TITLE
Updating the website with custom code

### DISCUSSION THREAD

**Discussion by Draise14** [Bforartists/Bforartists#3154] [Ideas]
I have been seeing some feedback from users that see the website and usually find it conflicting that the software work is so detailed and polished and great, but the website feels dated. Maybe because it's wordpress? 

There are some key things we don't have that could improve:
- Better responsive design
- Comparison Sliders and visuals
- Gif or animated illustration of feature benefits
- More visual heroes of the software, features and art showcasing the features 
- Maybe even an online and responsive design of the documentation

I do some web-design with Webflow and can export code that could be hosted anywhere - not CMS ready, but for static content I could give an oomph for later this year. I have been thinking long and hard about it - I'd like to do it one day. 

1. I just wanted to ask, what is the feasibility to host new pages/code for the website? 
2. Does everything need to be done with Wordpress? 
3. Could a site upgrade or replacement be uploaded/updated with FTP maybe? 
4. Or maybe would it make more sense to use a completely different editor with a new site made/hosted elsewhere and switch DNS? _I use Webflow, which is excellent - but if anything it's code can also be exported and hosted elsewhere for static content._ 
5. What kind of leeway for the web-design/publicity do we have for the near future? Would you be open to something new @ReinerBforartists eventually? 

**Plan of action:**
1. I can make a new website with a mockup, working responsively first as a proposal locally - 100% done.
2. You can check it out, see how it feels
3. Then we can worry later about how to make it work with the existing domain/seo/hosting etc. If you also like it, we can find a way to pass it over to your servers, etc. 

What do you think? This is a mid-range goal after perfecting the merging/weekly work of maintaining Bforartists.

**Comment by ReinerBforartists** [comment]
I am all in for improvements. But you have already a time problem. And i don't have the power anymore for a complete redesign of the page.

Wordpress is one of the easiest and fastest solutions out there. It is quick and ready to go in five minutes. And it allowed me to do the impossible. To create a community page with a user image gallery, a pm system and a board. And it is working. Other solutions requires lots of custom work for that. Months or even years of work. We both don't have this time.

You have access to the page. So feel free to do the required improvements that are possible. Animated gifs should not be a problem. Just take care that the page loads at the user side within a human time frame. More content means longer loading times. And Wordpress is already a resource hog.

And beware, a rewrite of the page from scratch will most probably kill you. It's too much workload. And the more customized it is the bigger is the burden to keep it working. These are the reasons to use Wordpress. Even when more elegant solutions are possible.

We don't have millions of users. And no pressure to make the most out of it. Most open source projects doesn't even have a own page. I personally think the page is fine as it is.

**Reply by Draise14** [reply]
Ah ok. Yeah, it's a mid to long term project. I can have an itern work it in my business eventually, but we don't use Wordpress for various reasons. Making a site in webflow also is.. quite fast, just finished 4 page website presentations for a client in the past couple of weeks. 

I feel a little held back by the editors of Wordpress probably by lack of use in the pipeline, so making changes with what is there seems to be a lot slower. But yeah, no rush. I'll put it on a drawing board, and yes, first priority is client work/business, Bforartists maintenance, then I can think about the rest later.

**Reply by Draise14** [reply]
Ah, the PM system and forums could be in Discord completely, and image galleries could be hero based or competition based - so maybe less is more as a website in the future.

**Comment by ReinerBforartists** [comment]
A simple static image gallery was never the problem. The culprit is the dynamic user image gallery. With upload at the front page by users. This will not work with Webflow. Webflow is meant for static pages. And a static page is done in a few minutes with pure html anyways. My first webpage ever was a static html page. I had written and maintained it in a stone old text only html editor back in 2000.

Users and dynamic content is where things becomes really complicated. That's where and why you need a cms. I have done quite a few pages in the last few years. And Wordpress turned out to be the fastest and easiest solution. And the most reliable one. My other wordpress page runs since many years, coming from Wordpress 3 or 4 i think. Must be more than 10 years now. And the migrations to the new versions were more or less trouble free. You might remember that we had a Drupal page in the past, based at Drupal 7. Drupal fell apart then. And we couldn't migrate to Drupal 8 since there was no official way for it. And even when, this would have been a downgrade. My Drupal love lasted a bit more than three years. I would never touch it again.

I also know the other obvious solutions like Contao, Typo3, October etc. I even tested to work with a PHP framework like Laravel. I've been there, and tried them all. And i have chosen Wordpress for a good reason then.

Keep your hands away from everything static, from exotic solutions, and from too much customization. And from non programming static html page builders like Webflow or Hugo. It will break your neck in the long run. This is my experience from the last 20 years with creating and maintaining my own webpages. Our current Bforartists webpage might not be the freshest anymore. And Wordpress is a beast. But it is working, as secure as it can be, future proof, and most of the times spam free. And maintained with a few clicks every month.

As told, you have access to the page. When you want to add a gif, then simply upload it and add it to the page text. Already this is ways more complicated in other solutions. To change the theme might be a bigger operation though. But even this is possible.

Discord is still a special chapter. I would not be happy to completely migrate the community to Discord. Even when de facto 99% already happens there. Discord is a data octopus. I would NEVER trust their pm system. Even after going through the privacy settings you will still give away lots of your privacy and your data sovereignty. Discord is a company, it needs to make money. It might change its terms of use tomorrow, or be quickly not available anymore. And then we still have a fallback strategy.

That's my five cents on the matter.
